src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        String colorOfFigure = colorSupplier.getRandomColor();
+        int numberOfFigure = random.nextInt(6);
+        if (numberOfFigure == 0) {
+            return new Circle(colorOfFigure, random.nextDouble());
+        }
+        if (numberOfFigure == 1) {
+            return new Square(colorOfFigure, random.nextDouble());
+        }
+        if (numberOfFigure == 2) {
+            return new RightTriangle(colorOfFigure, random.nextDouble(), random.nextDouble());
+        }
+        if (numberOfFigure == 3) {
+            return new Rectangle(colorOfFigure, random.nextDouble(), random.nextDouble());
+        }
+        if (numberOfFigure == 4) {
+            return new IsoscelesTrapezoid(colorOfFigure,
+                    random.nextDouble(), random.nextDouble(), random.nextDouble());
```

Comment:
passing expressions as constructor arguments decreases code readability

consider creating separate variables and passing them into constructor

see [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#%D1%81reating-a-figure-dont-pass-expressions-in-constructor)